### PR TITLE
docker-compose log dumps when verbose

### PR DESCRIPTION
**Pull request by tzaffi**
## Summary

To improve visibility of C.I. build failures, utilize new logging capabilities in the sandbox (https://github.com/algorand/sandbox/pull/181) when `VERBOSE_HARNESS=1`.

In particular under verbose, we can now see:
* traces of the `docker compose up` command
* dumped `docker-compose logs` for:
  * indexer-db
  * algod
  * indexer

### TODO

- [x] be deliberate about which branches are being pinned in `.env` before merging

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -2,7 +2,7 @@
 VERBOSE_HARNESS=0
 
 # Used to determine sandbox build type:
-TYPE="source" # OR "source"
+TYPE="source" # OR "channel"
```
Comment (tzaffi) [inline .env:5]:
```suggestion
TYPE="source" # OR "source"
```

File: .env [hunk 2]
```diff
@@ -17,12 +17,12 @@ NETWORK_TEMPLATE="images/algod/DevModeNetwork.json"  # refers to the ./images di
...
 NODE_ARCHIVAL="False"
-INDEXER_BRANCH="master"
-INDEXER_SHA=""
+INDEXER_BRANCH="hotfix/2.15.4"
+INDEXER_SHA="8ad75f37d71786df72163d2d0d7d2e0d1809a8e5"
 
 # Sandbox configuration:
 SANDBOX_URL="https://github.com/algorand/sandbox"
-SANDBOX_BRANCH="master"
+SANDBOX_BRANCH="verbose-harness"
```
Comment (tzaffi) [inline .env:25]: TODO: pin back to `master`'s before merging

File: scripts/up.sh [hunk 3]
```diff
@@ -71,6 +71,34 @@ pushd "$LOCAL_SANDBOX_DIR"
...
+  echo "$THIS: sandbox docker-compose logs indexer:"
+  ./sandbox dump indexer
+  echo "----------------------------------------"
+fi
 echo "$THIS: seconds it took to finish getting sandbox harness ($(pwd)) up and running: $(($(date "+%s") - START))s"
+set -e
+
+if [ "$error" -ne 0 ]; then
+  exit "$error"
+fi
```
Comment (tzaffi) [inline scripts/up.sh:104]:
```suggestion
fi

```

File: .env [hunk 4]
```diff
@@ -17,12 +17,12 @@ NETWORK_TEMPLATE="images/algod/DevModeNetwork.json"  # refers to the ./images di
 NETWORK_NUM_ROUNDS=30000
 INDEXER_URL="https://github.com/algorand/indexer"
 NODE_ARCHIVAL="False"
-INDEXER_BRANCH="master"
-INDEXER_SHA=""
+INDEXER_BRANCH="hotfix/2.15.4"
+INDEXER_SHA="8ad75f37d71786df72163d2d0d7d2e0d1809a8e5"
```
Comment (tzaffi) [inline .env:21]:
```suggestion
INDEXER_SHA=""
```

File: .env (hunk 1 above)
Comment (bbroder-algo) [reply .env:5]: wut

**Review by bbroder-algo — APPROVED** [review]

File: .env [hunk 5]
```diff
@@ -22,7 +22,7 @@ INDEXER_SHA=""
 
 # Sandbox configuration:
 SANDBOX_URL="https://github.com/algorand/sandbox"
-SANDBOX_BRANCH="master"
+SANDBOX_BRANCH="verbose-harness"
```
Comment (tzaffi) [inline .env:25]:
```suggestion
SANDBOX_BRANCH="master"
```

**Review by Eric-Warehime — APPROVED** [review]

**Review by winder — APPROVED** [review]
LGTM

**Pull request merged**